primitives/src/lib.rs

```diff
@@ -80,9 +88,31 @@ impl Default for CommitteeSeats {
     }
 }
 
+/// Configurable parameters for kick-out validator mechanism
+#[derive(Decode, Encode, TypeInfo, Debug, Clone, Copy, PartialEq, Eq)]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+pub struct CommitteeKickOutThresholds {
+    /// in a session, what is a limit for produced blocks under which the session counts as bad uptime
+    pub block_count_threshold: BlockCount,
```

Comment:
I think this is not the ideal way of defining a bad session. It would be better to make this a minimum fraction (percentage) of performance that is acceptable. Like `70%` means you must have >=70% performance in a session.

Absolute block counts are problematic, because they are sensitive to committee size etc. For instance, imagine we set the threshold to 50, forget about it and then increase committee size to 20 nodes. In which case we would ban all nodes :/

Reply:
See my proposal below